apiary-extensions-terraform/privileges-grantor/main.tf

```diff
@@ -0,0 +1,103 @@
+resource "aws_iam_role" "iam_for_privilege_grantor" {
+  name = "${local.instance_alias}_iam_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "privilege_grantor_lambda_vpc_access" {
+  name        = "${local.instance_alias}-lambda-vpc-access"
+  description = "VPC and CloudWatch access for Privilege Grantor Lambda function"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "ec2:CreateNetworkInterface",
+                "ec2:DescribeNetworkInterfaces",
+                "ec2:DeleteNetworkInterface"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "privilege_grantor_role_policy_attach" {
+  role       = "${aws_iam_role.iam_for_privilege_grantor.name}"
+  policy_arn = "${aws_iam_policy.privilege_grantor_lambda_vpc_access.arn}"
+}
+
+resource "aws_sqs_queue" "privilege_grantor_sqs_queue" {
+  name = "${local.instance_alias}-sqs-queue"
+}
+
+resource "aws_iam_role_policy" "sqs_for_privilege_grantor" {
+  name = "${local.instance_alias}-sqs-policy"
+  role = "${aws_iam_role.iam_for_privilege_grantor.id}"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": {
+        "Effect": "Allow",
+        "Action": [
+          "sqs:ReceiveMessage",
+          "sqs:DeleteMessage",
+          "sqs:GetQueueAttributes"
+        ],
+        "Resource": "${aws_sqs_queue.privilege_grantor_sqs_queue.arn}"
+    }
+}
+EOF
+}
+
+resource "aws_lambda_function" "privilege_grantor_fn" {
+  s3_bucket     = "${var.pg_lambda_bucket}"
+  s3_key        = "${var.pg_jars_s3_key}/apiary-privileges-grantor-lambda-${var.pg_lambda_version}.zip"
+  function_name = "${local.instance_alias}-fn"
+  role          = "${aws_iam_role.iam_for_privilege_grantor.arn}"
+  handler       = "com.expediagroup.apiary.extensions.events.metastore.consumer.privilegesgrantor.lambda.PrivilegesGrantorLambda::handleRequest"
+  runtime       = "java8"
+  memory_size   = "${var.memory}"
+  timeout       = "15"
+  publish       = true
+
+  environment {
+    variables = {
+      THRIFT_CONNECTION_URI = "${var.pg_metastore_uri}"
+    }
+  }
+
+  vpc_config {
+    subnet_ids         = ["${var.subnets}"]
+    security_group_ids = ["${var.security_groups}"]
+  }
+}
+
+resource "aws_lambda_event_source_mapping" "sqs_ecs_mapping" {
```

Comment:
maybe I miss this but where can we subscribe to particular events e.g. CREATE events only?

Reply:
Oh yes. I forgot about that part. I will add the message attributes filter to the queue.

Reply:
I have added the filter.